A lot of suggested changes to link.md

e-lo opened the pull request:
1. moved a lot of things to be optional, thinking about the information
   we have from surveys, we might not have info on specific times.
2. changed several variable names to be more intuitive (to me) and not
   have spaces.
3. Updated descriptions of variables to be slightly more generic. PLEASE CHECK.

e-lo committed:
A lot of suggested changes.

1. moved a lot of things to be optional, thinking about the information
we have from surveys, we might not have info on specific times.
2. changed several variable names to be more intuitive (to me) and not
have spaces.
3. Updated descriptions of variables to be slightly more generic.

lmz commented:
I think the linknum field (or some rename of it) should stay for the links file, because otherwise the order is implied by the ordering in the file and I prefer explicit over implicit.

I also think keeping A_seq and B_seq in mandatory may be a good idea for when the stop_id is ambiguous (the transit route hits the same stop_id multiple times).

e-lo commented:
- Agree on `link_num` since that can be derived from any path file.
- A_seq and B_seq we might not have (probably wouldn't have) for any observed data with any degree of certainty since we aren't completely sure about anything that happens outside of when the passenger is on board.  Thoughts?

e-lo committed:
Re-add link_num; rename to link_XX

- fixes inadvertent drop of linknum variable
- renames linkxxx variables to link_xxx to be consistent with how other
variables named

lmz commented:
Good point for A_seq and B_seq.   Ok, LGTM!
